addons/mrp/report/mrp_report_mo_overview.py

```diff
@@ -100,6 +100,21 @@ def _get_mo_summary(self, production, components, operations):
         }
 
     def _format_state(self, record):
+        if record._name == 'mrp.production' and record.state in ('draft', 'confirmed') and record.move_raw_ids:
+            producibles = [
+                float_round(
+                    (move.reserved_availability + move.product_id.uom_id._compute_quantity(move.product_id.free_qty, move.product_uom))\
+                    / (record.bom_id.product_qty or (move.product_qty / record.product_uom_qty)),
```

Comment:
This is not correct, it has several issues:
1. no matter you use bom or not, the number on the MO can be different from the bom, so you shouldn't rely on the qty on bom_id.
2. even if you need the qty on bom, it should be the qty from bom.line instead of bom_id.product_qty.
3. the divident is in move.product_uom while the divisor is in product's default uom (second half).

Reply:
Hello @yhu-odoo @ticodoo 
we use `bom_id` as per MGM review https://www.awesomescreenshot.com/video/19132376?key=393fcbb65ff56d9a53c78b0cfa3d32c3
if video behaviour is fine and no need to handle such situations then we can remove `bom_id`.
Thanks

Reply:
Yes I know need some improvement in math calculation but in BoM while calculating the producible qty it uses BoM product Quantity
here https://github.com/odoo/odoo/blob/master/addons/mrp/report/mrp_report_bom_structure.py#L31

Ex. 1
```
BOM_Quantity = 2
producibles = [9.0,4.0]
min(producibles) * BOM_Quantity # output -> 8
```

Ex. 2
```
BOM_Quantity = 1
producibles = [9.0,4.0]
min(producibles) * BOM_Quantity # output -> 4
```
see https://excalidraw.com/#json=N32R1Lvgy1S7VAaFguFeR,xAMy6YVGjo0aCirAl33Ywg

Let me know if I am wrong or misunderstood.

Reply:
I am discussing with mgm. He did indeed mean what you were thinking (use the BoM qty), but we are debating if it makes sense (I personally think it is confusing for a number of reasons). I will let you know the end decision

Reply:
Ok, I win :trophy: The calculation should be based on the MO qty because:
- the producible_qty will always be 0 if the MO qty to produce is less than the BoM qty
- backorders' producible_qty values may not add up correctly with the original MO's in case of qtys not done in multiples of the BoM qty
- users don't see the BoM qty in the MO, so the producible_qty may be incredibly confusing since they can't see the math that calculates it

Reply:
You still haven't fixed issue #3

Reply:
@ticodoo  I have applied your suggested changes but also I have modified similar to BOM logic.

Reply:
You are still dividing in the wrong uom (i.e. problem 3 that yhu pointed out a MONTH AGO)